balancer-js/src/modules/sor/pool-data/onChainData.ts

```diff
@@ -141,7 +142,7 @@ export async function getOnChainBalances<
         'getSwapFeePercentage'
       );
     }
-    if (pool.poolType.toString() === 'GyroE' && pool.poolTypeVersion === 2) {
+    if (['GyroE', 'Gyro2', 'Gyro3'].includes(pool.poolType.toString())) {
```

Comment:
Not sure you've thought about this change? Check out the contracts for these pool types and see if the function exists?